app/models/Arrival.scala

```diff
@@ -16,97 +16,27 @@
-        case ArrivalRejection =>
-          if (messagesMetaData.count(_.messageType == ArrivalNotificationSubmitted) > messagesMetaData.count(_.messageType == ArrivalRejection)) {
-            ArrivalNotificationSubmitted
-          } else {
-            ArrivalRejection
-          }
-        case UnloadingRemarksRejection =>
-          if (messagesMetaData.count(_.messageType == UnloadingRemarksSubmitted) > messagesMetaData.count(_.messageType == UnloadingRemarksRejection)) {
-            UnloadingRemarksSubmitted
-          } else {
-            UnloadingRemarksRejection
-          }
-        case XMLSubmissionNegativeAcknowledgement if messagesMetaData.count(_.messageType == UnloadingRemarksSubmitted) >= 1 =>
-          if (
-            messagesMetaData.count(_.messageType == UnloadingRemarksSubmitted) > messagesMetaData.count(_.messageType == XMLSubmissionNegativeAcknowledgement)
-          ) {
-            UnloadingRemarksSubmitted
-          } else {
-            XMLSubmissionNegativeAcknowledgement
-          }
-        case XMLSubmissionNegativeAcknowledgement =>
-          if (
-            messagesMetaData
-              .count(_.messageType == ArrivalNotificationSubmitted) > messagesMetaData.count(_.messageType == XMLSubmissionNegativeAcknowledgement)
-          ) {
-            ArrivalNotificationSubmitted
-          } else {
-            XMLSubmissionNegativeAcknowledgement
-          }
-        case value => value
-      }
-    }
-  }
-
-  def previousStatus: ArrivalStatus = {
-
-    implicit val localDateOrdering: Ordering[LocalDateTime] = _ compareTo _
-
-    val previousMessage = messagesMetaData.sortBy(_.dateTime).takeRight(2).head
-
-    val messagesWithSameDateTime = messagesMetaData.filter(_.dateTime == previousMessage.dateTime)
-
-    if (messagesWithSameDateTime.size == 1) {
-      previousMessage.messageType
-    } else {
-
-      currentStatus match {
-        case ArrivalStatus.XMLSubmissionNegativeAcknowledgement =>
-          if (previousMessage.messageType == ArrivalNotificationSubmitted | previousMessage.messageType == UnloadingRemarksSubmitted) {
-            previousMessage.messageType
-          } else {
-            messagesWithSameDateTime.map(_.messageType).max
-          }
-        case _ => messagesWithSameDateTime.map(_.messageType).max
-      }
-    }
-  }
-}
+                   movementReferenceNumber: String,
+                   status: ArrivalStatus,
+                   previousStatus: Option[ArrivalStatus] = None
+)
 
 object Arrival {
 
-  implicit val reads: Reads[Arrival] = Json.reads[Arrival]
+  implicit val reads: Reads[Arrival] = (
+    (__ \ "arrivalId").read[ArrivalId] and
+      (__ \ "created").read[LocalDateTime] and
+      (__ \ "updated").read[LocalDateTime] and
+      (__ \ "movementReferenceNumber").read[String] and
+      (__ \ "status").read[ArrivalStatus] and
+      (__ \ "previousStatus").readNullable[ArrivalStatus]
+  )(Arrival.apply _)
```

Comment:
Would this not just do the same thing as `implicit val reads: Reads[Arrival] = Json.reads[Arrival]`?

Reply:
Yep, fixed that